zookeeper/core/component.py

```diff
@@ -277,6 +277,54 @@ def wrapped_fn(instance) -> List[str]:
     component_cls.__dir__ = wrapped_fn
 
 
+def _wrap_configure(component_cls: Type) -> None:
+    if hasattr(component_cls, "__configure__"):
+        if not callable(component_cls.__configure__):
+            raise TypeError(
+                "The `__configure__` attribute of a @component class must be a "
+                "method."
+            )
+        call_args = inspect.signature(component_cls.__configure__).parameters
+        configure_args = inspect.signature(configure).parameters
+
+        error_message = (
+            "The `__configure__` method of a @component class must match the arguments "
+            f"of `configure()`, but `{component_cls.__name__}.__configure__` "
+            f"accepts arguments {tuple(name for name in call_args)}. Valid arguments: "
+            f"({', '.join(name for name in configure_args)})"
+        )
+
+        for arg_name, arg_param in call_args.items():
+            if (
+                arg_param.kind in (arg_param.VAR_POSITIONAL, arg_param.VAR_KEYWORD)
+                or arg_name in configure_args
+                or arg_name == "self"
+            ):
+                continue
+            raise TypeError(error_message)
```

Comment:
Could we inline the `error_message` here since now it's only used once? Also code-cov is warning that this line apparently isn't being tested, is that right?

Reply:
No it isn't, this error is tested like 3 times

Reply:
I don't think that's the case, I don't see a match against `pytest.raises` with that specific error message?

Reply:
Huh yeah that's a different one... I'll update this sometime then, I'm no longer waiting for it anyway